sensor/common/sensor/central_sender_impl.go

```diff
@@ -112,9 +110,20 @@ func (s *centralSenderImpl) send(stream central.SensorService_CommunicateClient,
 			}
 
 			if msg.GetEvent().GetSynced() != nil {
-				log.Info("Sending synced signal to Central")
-				if s.onSync != nil {
-					go s.onSync()
+				unchangedIds := s.observationSet.Close()
+				// Enhance sync with all the observed IDs
+				msg.GetEvent().GetSynced().UnchangedIds = unchangedIds
+				log.Infof("Sending synced signal to Central. Adding %d events as unchanged", len(unchangedIds))
+
+				// Set metrics count for unchanged IDs
+				for i, key := range unchangedIds {
+					parts := strings.SplitN(key, ":", 2)
```

Comment:
This section seems unnecessary? just inc by the len(of unchanged ids) and print out the ideas using %+v?

Reply:
Removed